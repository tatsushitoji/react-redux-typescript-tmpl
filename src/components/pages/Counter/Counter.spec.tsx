import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Counter } from '.';

let mockIncrement: any;
let mockDecrement: any;
let mockAsyncIncrement: any;

beforeEach(() => {
  mockIncrement = jest.fn();
  mockDecrement = jest.fn();
  mockAsyncIncrement = jest.fn();
});

test('should render self with initial state', () => {
  const tree = shallow(
    <Counter
      count={0}
      increment={mockIncrement}
      decrement={mockDecrement}
      asyncIncrement={mockAsyncIncrement}
    />,
  );
  expect(toJson(tree)).toMatchSnapshot();
});
