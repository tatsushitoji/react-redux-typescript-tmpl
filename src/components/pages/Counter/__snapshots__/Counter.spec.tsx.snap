// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render self with initial state 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Counter
    </title>
  </HelmetWrapper>
  <CounterPage
    asyncIncrement={[MockFunction]}
    count={0}
    decrement={[MockFunction]}
    increment={[MockFunction]}
  />
</Fragment>
`;
